NitroxClient/GameLogic/Spawning/InstalledBatteryEntitySpawner.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using NitroxClient.Communication;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures.GameLogic.Entities;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Packets;
+using NitroxModel_Subnautica.DataStructures;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning;
+
+public class InstalledBatteryEntitySpawner : EntitySpawner<InstalledBatteryEntity>
+{
+    public override IEnumerator SpawnAsync(InstalledBatteryEntity entity, TaskResult<Optional<GameObject>> result)
+    {
+        Optional<GameObject> parent = NitroxEntity.GetObjectFrom(entity.ParentId);
+
+        if (!parent.HasValue)
+        {
+            Log.Info($"Unable to find parent to install battery {entity}");
+            result.Set(Optional.Empty);
+        }
+
+        EnergyMixin energyMixin = parent.Value.GetComponent<EnergyMixin>();
+
+        if (!energyMixin)
+        {
+            Log.Info($"Unable to find EnergyMixin on parent to install battery {entity}");
+            result.Set(Optional.Empty);
```

Comment:
And here. Though if this check doesn't return the lines below will error accessing energymixin

Reply:
done